lib/plug/conn.ex

```diff
@@ -829,6 +838,9 @@ defmodule Plug.Conn do
 
     %{conn | private: private}
   end
+
+  defp valid_header_key?(""), do: false
+  defp valid_header_key?(key), do: String.downcase(key) == key
```

Comment:
We need this to be fast. We can remove the first clause and we can assume ASCII only, so just traverse the binary and check if any chars are in ?A..?Z.